### PR TITLE
Upgrade eks-operator chart to 1.0.6

### PR DESCRIPTION
**Pull request by thedadams**
These changes include support for launch templates and spot instances for managed EKS clusters.

eks-operator PR:
https://github.com/rancher/eks-operator/pull/27

Issue:
https://github.com/rancher/rancher/issues/28803

### CODE REVIEW

File: charts/rancher-eks-operator/1.0.600/Chart.yaml [hunk 1]
```diff
@@ -4,5 +4,5 @@ description: A Helm chart for provisioning EKS clusters
 home: https://github.com/rancher/eks-operator
 sources:
   - "https://github.com/rancher/eks-operator"
-version: 1.0.500
-appVersion: v1.0.5
+version: 1.0.600
+appVersion: v1.0.6-rc1
```
Comment (rmweir) [inline charts/rancher-eks-operator/1.0.600/Chart.yaml:8]: has 1.0.500 already been in a release?

File: charts/rancher-eks-operator/1.0.600/templates/eksclusterconfig.yaml [hunk 2]
```diff
@@ -2,8 +2,6 @@ apiVersion: apiextensions.k8s.io/v1beta1
 kind: CustomResourceDefinition
 metadata:
   name: eksclusterconfigs.eks.cattle.io
-  annotations:
-    "helm.sh/resource-policy": keep
```
Comment (rmweir) [inline charts/rancher-eks-operator/1.0.600/templates/eksclusterconfig.yaml]: why is this being removed?

**Review by rmweir — CHANGES_REQUESTED** [review]

File: charts/rancher-eks-operator/1.0.600/Chart.yaml (hunk 1 above)
Comment (thedadams) [reply charts/rancher-eks-operator/1.0.600/Chart.yaml:8]: Yes, I believe it was in 2.5.4 for private endpoint stuff.

File: charts/rancher-eks-operator/1.0.600/templates/eksclusterconfig.yaml (hunk 2 above)
Comment (thedadams) [reply charts/rancher-eks-operator/1.0.600/templates/eksclusterconfig.yaml]: I ran go generate in `eks-operator` and copied this over.

File: charts/rancher-eks-operator/1.0.600/Chart.yaml (hunk 1 above)
Comment (cbron) [reply charts/rancher-eks-operator/1.0.600/Chart.yaml:8]: This is easy to check by looking at release branch, it was added: https://github.com/rancher/system-charts/tree/release-v2.5/charts/rancher-eks-operator

**Review by cbron — APPROVED** [review]
As long as instance type isn't needed anymore it lgtm, would like Rickys approval also

**Review by rmweir — APPROVED** [review]
LGTM

**Pull request merged**